handlers/handlers.go

```diff
@@ -46,74 +55,284 @@ func parseReqBody(w http.ResponseWriter, body io.ReadCloser, bodyObj *models.Pre
 	return nil
 }
 
+func parseStringToInt(strings ...string) ([]int, error) {
+	var (
+		val int
+		err error
+	)
+
+	vals := make([]int, 0)
+	for _, str := range strings {
+		val = 0
+		if str != "" {
+			val, err = strconv.Atoi(str)
+			if err != nil {
+				errMsg := "failed to convert string to int: " + err.Error()
+				return nil, errors.New(errMsg)
+			}
+		}
+		vals = append(vals, val)
+	}
+
+	return vals, nil
+}
+
 // PostPrefsHandler creates new preferences for a user
-func PostPrefsHandler(w http.ResponseWriter, r *http.Request) {
+func (env *Env) PostPrefsHandler(w http.ResponseWriter, r *http.Request) {
 	defer r.Body.Close()
 
 	reqBody := models.NewPreferences()
 	if err := parseReqBody(w, r.Body, reqBody); err != nil {
 		return
 	}
 
+	vals, err := parseStringToInt(r.Header.Get("User-ID"))
+	if err != nil {
+		log.Println(err.Error())
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	reqBody.UserID = vals[0]
+
+	if err = env.DB.CreatePrefs(reqBody); err != nil {
+		errMsg := fmt.Sprintf(
+			"failed to create prefs (%+v): %s",
+			*reqBody,
+			err.Error(),
+		)
+		log.Println(errMsg)
+		responseCode := http.StatusInternalServerError
+		if err == models.ErrPrefsExists {
+			responseCode = http.StatusConflict
+		}
+		http.Error(w, errMsg, responseCode)
+		return
+	}
+
+	reqBody.UserID = 0
+
+	w.Header().Set("Content-Type", "application/json")
 	json.NewEncoder(w).Encode(reqBody)
 }
 
-// GetPrefsHandler gets filtered preferences for a user
-func GetPrefsHandler(w http.ResponseWriter, r *http.Request) {
-	queryValues, err := url.ParseQuery(r.URL.RawQuery)
+// PostPrefsConvHandler creates new conversation preferences for a user
+func (env *Env) PostPrefsConvHandler(w http.ResponseWriter, r *http.Request) {
+	defer r.Body.Close()
+
+	reqBody := models.NewConversationPrefs()
+	if err := parseReqBody(w, r.Body, reqBody); err != nil {
+		return
+	}
+
+	vals, err := parseStringToInt(r.Header.Get("User-ID"))
 	if err != nil {
-		errMsg := "Failed to parse query: " + err.Error()
+		log.Println(err.Error())
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	if err := env.DB.CreatePrefsConv(vals[0], reqBody); err != nil {
+		errMsg := fmt.Sprintf(
+			"failed to create conversation prefs (%+v): %s",
+			*reqBody,
+			err.Error(),
+		)
 		log.Println(errMsg)
-		http.Error(w, errMsg, http.StatusBadRequest)
+		responseCode := http.StatusInternalServerError
+		if err == models.ErrPrefsExists {
+			responseCode = http.StatusConflict
+		}
+		http.Error(w, errMsg, responseCode)
+		return
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	json.NewEncoder(w).Encode(reqBody)
```

Comment:
Same as above. This should maybe return `201 Created` instead of `200 OK`.

Reply:
Same as above as well.